posts/2023-02-27-introducing-new-conda.md

```diff
@@ -0,0 +1,53 @@
+
+On installation or update, a symlink to the launcher script corresponding to every command within the conda environment is created in a separate scripts directory. The launcher script will determine which command needs to be run from the conda environment based on the name of the symlink. For example, running the `jupyter` symlink in the `cms_conda_scripts/analysis3-23.01.d/bin/` directory is the equivalent of running `cms_conda/envs/analysis3-23.10/bin/jupyter` from within the container. The launcher script can also run arbitrary commands. For example, to launch a shell within the container, run `launcher.sh bash`. It is this scripts directory that is placed in `$PATH`, and launches Singularity only when necessary. The downside of this is that `singularity` can be expensive to launch. If your workflow requires launching applications from the `analysis3` environment in quick succession, it may be worth re-factoring the workflow such that it runs from within the container using `launcher.sh`. The typical `analysis3` workflow comprising python scripts or jupyter notebooks is not affected by this, and can be run as normal.
+
+## Key differences between `conda_concept` and `conda`
+The key difference of the approach used by the `conda_concept` environments is the use of singularity's ability to manage squashfs file systems. A squashfs can be thought of a something like a tar file that can be accessed as if it were an entirely new file system. The obvious advantage of placing a conda environment in a squashfs is that it reduces the file count of the entire environment on `/g/data` to one. The drawback of using singularity to manage the squashfs is that containers have a number of restrictions placed on them for security reasons, (e.g. the `newgrp`, `switchproj` and `qcat` commands cannot be run from inside a container). Because of this, once the module is loaded, the user is kept out of the containerised environment unless a command that exists inside the container is run. This is accomplished by the use of a 'launcher' script that runs `singularity` and executes the command from within the container. For more details on this, see the [CMS Wiki page](https://climate-cms.org/cms-wiki/resources/resources-conda-setup.html). This leads to known issues with the environment. 

```

Comment:
```suggestion
The key difference of the approach used by the `conda_concept` environments is the use of singularity's ability to manage squashfs file systems. A squashfs can be thought of a something like a tar file that can be accessed as if it were an entirely new file system. The obvious advantage of placing a conda environment in a squashfs is that it reduces the file count of the entire environment on `/g/data` to one. 
The drawback of using singularity to manage the squashfs is that containers have a number of restrictions placed on them for security reasons, (e.g. the `newgrp`, `switchproj` and `qcat` commands cannot be run from inside a container). Because of this, once the module is loaded, the user is kept out of the containerised environment unless a command that exists inside the container is run. This is accomplished using a 'launcher' script that runs `singularity` and executes the command from within the container. For more details on this, see the [CMS Wiki page](https://climate-cms.org/cms-wiki/resources/resources-conda-setup.html). This leads to known issues with the environment. 

```